agent/taskresource/types/types_test.go

```diff
@@ -74,3 +73,17 @@ func TestMarshalUnmarshalSSMSecretResource(t *testing.T) {
 	assert.Equal(t, resourcestatus.ResourceCreated, ssmRes.GetDesiredStatus())
 	assert.Equal(t, resourcestatus.ResourceRemoved, ssmRes.GetKnownStatus())
 }
+
+func TestMarshalUnmarshalASMSecretResource(t *testing.T) {
```

Comment:
it seem that this test only tests Unmarshal? same for TestMarshalUnmarshalSSMSecretResource